onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc

```diff
@@ -201,11 +200,16 @@ SubGraphCollection_t RemoveEmptyShapeNodes(const onnxruntime::GraphViewer& graph
         for (const auto& dim : input_shape->dim()) {
           std::string dim_name = dim.dim_param();
           if (!dim_name.empty()) {
-            if ((dim_name.find(exclude_dim_name1) != std::string::npos) || (dim_name.find(exclude_dim_name2) != std::string::npos)) {
-              exclude_node = true;
+            for (const auto& exclude : exclude_dim_name) {
```

Comment:
use unordered_set instead of iterating through vector.

Reply:
Unordered_set won't save us time as it's not searching string in the set. Instead exclude_dim_name only contains key words and we search each key word in the input names. If any key word is found in the input name, we know the input could have empty shape.

Reply:
I see. I misread it. The logic is extremely model specific then.

Reply:
Yes. There might be a better way to do this, but this is a quick workaround and will not cover all models and all cases. We need TensorRT new release to support empty shape as the final solution.